internal/alb/tg/targetgroup.go

```diff
@@ -150,6 +156,33 @@ func (controller *defaultController) reconcileTGInstance(ctx context.Context, in
 	return instance, nil
 }
 
+// resolveServiceHealthCheckPort checks if the service-port annotation is a string. If so, it tries to look up a port with the same name
+// on the service and use that port's NodePort as the health check port.
+func resolveServiceHealthCheckPort(store store.Storer, namespace string, serviceName string, servicePortAnnotation intstr.IntOrString) (* string) {
```

Comment:
1. I'd prefer to make this an member function of controller, so we don't need to pass singleton object(store) around.
1. I'd prefer to always return string instead of *string, when there is no "optional" semantic. (Ideally, every *xxx needs to be null-checked, otherwise, it should be xxx).